yunwoo/baekjoon/DFS-BFS/2667-단지번호붙이기.js

```diff
@@ -0,0 +1,72 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const n = Number(input[0]);
+  const graph = input.slice(1).map((el) => el.split("").map(Number));
+  const result = [];
+
+  const getIsValid = (y, x) => {
+    if (x >= 0 && y >= 0 && x < n && y < n && graph[y][x]) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const bfs = (y, x) => {
+    const dx = [0, 0, -1, 1];
+    const dy = [1, -1, 0, 0];
+    const queue = [[y, x]];
+    let count = 1;
+
+    graph[y][x] = 0;
+
+    while (queue.length) {
+      const [currentY, currentX] = queue.shift();
+
+      for (let i = 0; i < 4; i++) {
+        const nx = currentX + dx[i];
+        const ny = currentY + dy[i];
+
+        if (getIsValid(ny, nx)) {
+          graph[ny][nx] = 0;
+          queue.push([ny, nx]);
+
+          count++;
+        }
+      }
+    }
+
+    return count;
+  };
```

Comment:
```suggestion
  const moves = [
    [0, 1],
    [1, 0],
    [-1, 0],
    [0, -1],
  ];
  const bfs = (y, x) => {
    const queue = [[y, x]];
    let count = 1;

    graph[y][x] = 0;

    while (queue.length) {
      const [currentY, currentX] = queue.shift();
      
      moves.forEach(move => {
        const [dy, dx] = move;
        const [ny, nx] = [currentY + dy, currentX + dx];
        
        if (getIsValid(ny, nx)) {
          graph[ny][nx] = 0;
          queue.push([ny, nx]);

          count++;
        }
      });
    }

    return count;
  };
```
- dx dy배열을 하나의 2차원 배열인 moves로 관리하면 forEach를 활용한 함수형으로 작성도 가능하니 참고부탁드립니다

Reply:
- 와 훨씬 좋은 방법인거 같습니다!!! 👍